ethers-middleware/src/policy.rs

```diff
@@ -113,3 +95,95 @@ where
             .map_err(PolicyMiddlewareError::MiddlewareError)
     }
 }
+
+/// A `Policy` that only lives in memory
+#[derive(Debug)]
+pub struct MemoryPolicy {
+    pub rules: Vec<Rule>,
+}
+
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+impl Policy for MemoryPolicy {
+    type Error = PolicyError;
+
+    async fn ensure_can_send(&self, tx: TypedTransaction) -> Result<TypedTransaction, Self::Error> {
+        self.rules
+            .iter()
+            .find_map(|rule| rule.ensure(&tx).err())
+            .map_or(Ok(tx), Err)
+    }
+}
+
+/// Different rules that can be configured.
+#[derive(Debug)]
+pub enum Rule {
+    ReceiverAllowList(HashSet<Address>),
+    ReceiverBlockList(HashSet<Address>),
+    SenderAllowList(HashSet<Address>),
+    SenderBlockList(HashSet<Address>),
+    AllowList(HashSet<(Address, Address)>),
+    BlockList(HashSet<(Address, Address)>),
+    ValueCap(U256),
+    SenderValueCap(HashMap<Address, U256>),
+    ReceiverValueCap(HashMap<Address, U256>),
+    InvalidSelector(HashSet<Selector>),
+    InvalidReceiverSelector(HashMap<Address, Selector>),
+    // Other(Box<dyn Fn(&TypedTransaction) -> Result<(), PolicyError> + Send + Sync>)
```

Comment:
If you figure out a way to get this working, it's worth turning `ensure` to `async`, and making the Other variant be an async function. That way, we could even contact 3rd party service providers (or policy engines), and even read any expensive rules (e.g. ones which may need a lot of policy info from disk) concurrently. h/t @prestwich for this suggestion